File: Examples/test-suite/csharp/char_binary_runme.cs
using System;
using char_binaryNamespace;

public class char_binary_runme {

  public static void Main() {
    Test t = new Test();
    string hile = "hile";
    string hil0 = "hil\0";
    if (t.strlen(hile) != 4)
      throw new Exception("bad multi-arg typemap");

    if (t.strlen(hil0) != 4)
      throw new Exception("bad multi-arg typemap");

    if (t.ustrlen(hile) != 4)
      throw new Exception("bad multi-arg typemap");

    if (t.ustrlen(hil0) != 4)
      throw new Exception("bad multi-arg typemap");

    SWIGTYPE_p_char pc = char_binary.new_pchar(5);
    char_binary.pchar_setitem(pc, 0, 'h');
    char_binary.pchar_setitem(pc, 1, 'o');
    char_binary.pchar_setitem(pc, 2, 'l');
    char_binary.pchar_setitem(pc, 3, 'a');
    char_binary.pchar_setitem(pc, 4, '\0');

    /* FIXME: pc is NOT a string
    if (t.strlen(pc) != 4)
      throw new Exception("bad multi-arg typemap");
    if (t.ustrlen(pc) != 4)
      throw new Exception("bad multi-arg typemap");
    */

    /* FIXME: pc is NOT a string
    char_binary.var_pchar = pc;
    */
    char_binary.var_pchar = "hola";
    if (char_binary.var_pchar != "hola")
      throw new Exception("bad pointer case");

    /* FIXME: pc is NOT a string
    char_binary.var_namet = pc;
    */
    char_binary.var_namet = "hola";
    if (char_binary.var_namet != "hola")
        throw new Exception("bad pointer case");

    char_binary.delete_pchar(pc);
  }
}

File: Examples/test-suite/lua/char_binary_runme.lua
require("import")
require("char_binary")
v = char_binary
catch_undef_globs() -- catch "undefined" global variables

t = v.Test()
assert(t:strlen("hile") == 4, "bad multi-arg typemap")
assert(t:ustrlen("hile") == 4, "bad multi-arg typemap")
assert(t:strlen("hil\0") == 4, "bad multi-arg typemap")
assert(t:ustrlen("hil\0") == 4, "bad multi-arg typemap")

-- creating a raw char*
pc = v.new_pchar(5)
v.pchar_setitem(pc, 0, 'h')
v.pchar_setitem(pc, 1, 'o')
v.pchar_setitem(pc, 2, 'l')
v.pchar_setitem(pc, 3, 'a')
v.pchar_setitem(pc, 4, 0)

-- FIXME: strlen except 'char const *' but pc is 'char *'
if false then
  assert(t:strlen(pc) == 4, "bad multi-arg typemap")
  assert(t:ustrlen(pc) == 4, "bad multi-arg typemap")
end
-- FIXME: expected 'pchar' got 'char *'
if false then
  v.var_pchar = pc
else
  v.var_pchar = "hola"
end
assert(v.var_pchar == "hola", "bad pointer case")

-- FIXME: expected 'pchar' got 'char *'
if false then
  v.var_namet = pc
else
  v.var_namet = "hola"
end
assert(v.var_namet == "hola", "bad pointer case")

v.delete_pchar(pc)

